crates/apub/src/activities/voting/vote.rs

```diff
@@ -68,12 +68,23 @@ impl ActivityHandler for Vote {
 
     check_bot_account(&actor.0)?;
 
-    let enable_downvotes = LocalSite::read(&mut context.pool())
-      .await
-      .map(|l| l.enable_downvotes)
+    // Check for enabled federation votes
+    let local_site = LocalSite::read(&mut context.pool()).await;
+    let enable_federated_downvotes = local_site
+      .as_ref()
+      .map(|l| l.enable_downvotes && !l.reject_federated_downvotes)
       .unwrap_or(true);
-    if self.kind == VoteType::Dislike && !enable_downvotes {
-      // If this is a downvote but downvotes are ignored, only undo any existing vote
+
+    let enable_federated_upvotes = local_site
+      .as_ref()
+      .map(|l| !l.reject_federated_upvotes)
+      .unwrap_or(true);
+
+    let reject_vote_check = (self.kind == VoteType::Dislike && !enable_federated_downvotes)
+      || (self.kind == VoteType::Like && !enable_federated_upvotes);
+
+    if reject_vote_check {
+      // If this is a rejection, undo the vote
       match object {
         PostOrComment::Post(p) => undo_vote_post(actor, &p, context).await,
```

Comment:
@Nutomic One reason that people gave for rejecting votes, is that they'd like to decrease the DB load, and avoid federated vote inserts entirely. But L89 seems like it inserts, then removes these votes (which would fire plenty of DB triggers also). Wouldn't it be better to just not insert them in the first place?

Reply:
No this logic is used when a user first upvotes and then downvotes a post. So we cant apply the downvote, but leaving it as upvoted doesnt make sense either. Instead we remove the vote entirely. You can clarify the comments.

Reply:
Does that mean I need to add this check elsewhere also?

Reply:
Most of the logic is in mod.rs so it may be easier to make these changes there. For federation this should be all, of course you still need to add api checks.

Reply:
Regarding db writes, my assumption is that trying to delete nonexistent row doesnt write anything.